corporate/lib/stripe.py

```diff
@@ -155,6 +156,14 @@ def get_latest_seat_count(realm: Realm) -> int:
     return get_seat_count(realm, extra_non_guests_count=0, extra_guests_count=0)
 
 
+@cache_with_key(lambda realm: get_realm_seat_count_cache_key(realm.id), timeout=3600 * 24)
+def get_cached_seat_count(realm: Realm) -> int:
+    # This is a cache value  we're intentionally okay with not invalidating.
+    # All that means is that this value will lag up to 24 hours before getting updated.
+    # We use this for calculating the uploaded files storage limit for paid Cloud organizations.
+    return get_latest_seat_count(realm)
```

Comment:
I think `do_create_user`, `do_reactivate_user` and `do_deactivate_user` should be the sufficient list of places that would need to invalidate this cache - but upload storage limit feels like the kind of thing that makes sense to be lagged by up to 24h and might be worth the simplicity of not worrying about cache invalidation?

Reply:
I think the callers of `realm_user_count_by_role` would be the right set -- so basically what you mentioned plus the role-change code path, for users switching between guests and not. Could consider changing `maybe_enqueue_audit_log_upload` to have a more general name like `process_realm_license_count_change` and do this cache flush also. Could imagine having that accept the audit log event type and do the `RealmAuditLog.objects.create` as well.

Reply:
Opened https://github.com/zulip/zulip/issues/29758 to track this